homeassistant/components/mjpeg/camera.py

```diff
@@ -118,6 +119,14 @@ def __init__(
         if device_info is not None:
             self._attr_device_info = device_info
 
+    async def stream_source(self) -> str:
+        """Return the stream source."""
+        url = URL(self._mjpeg_url)
+        if self._username and self._password:
+            url = url.with_user(self._username)
+            url = url.with_password(self._password)
```

Comment:
```suggestion
        if self._username:
            url = url.with_user(self._username)
        if self._password:            
            url = url.with_password(self._password)
```

Reply:
Looks like its possible to have a password without a username from the schema above

```python
                username=entry.options.get(CONF_USERNAME),
                password=entry.options[CONF_PASSWORD],
```